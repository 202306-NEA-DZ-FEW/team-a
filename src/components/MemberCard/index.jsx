import Image from "next/image";
import Link from "next/link";
import React, { useEffect, useState } from "react";
import { AiFillLinkedin } from "react-icons/ai";
import { FaGithub } from "react-icons/fa";

import styles from "./styles.module.css";

function MemberCard({ teamMembers = [] }) {
  const [activeContent, setActiveContent] = useState(null);

  useEffect(() => {
    const imgBx = document.querySelectorAll(`.${styles.imgbx}`);

    const handleMouseOrTouch = (index) => {
      setActiveContent(index + 1);
    };

    imgBx.forEach((element, index) => {
      element.addEventListener("mouseenter", () => handleMouseOrTouch(index));
      element.addEventListener("click", () => handleMouseOrTouch(index));
      element.addEventListener("touchstart", (event) => {
        event.preventDefault();
        handleMouseOrTouch(index);
      });
    });

    return () => {
      imgBx.forEach((element, index) => {
        element.removeEventListener("mouseenter", () =>
          handleMouseOrTouch(index)
        );
        element.removeEventListener("click", () => handleMouseOrTouch(index));
        element.removeEventListener("touchstart", () =>
          handleMouseOrTouch(index)
        );
      });
    };
  }, []);

  return (
    <div className={styles.container}>
      <div className={styles.icon}>
        {teamMembers.map((member, index) => (
          <div
            key={index}
            className={`${styles.imgbx} ${
              index + 1 === activeContent ? styles.active : ""
            }`}
            style={{ "--i": index + 1 }}
          >
            <Image
              src={member.imageUrl}
              alt={`recoded${index + 1}`}
              className={styles.Image}
            />
          </div>
        ))}
      </div>
      <div className={styles.content}>
        {teamMembers.map((member, index) => (
          <div
            key={index}
            className={`${styles.contentBx} ${
              index + 1 === activeContent ? styles.active : ""
            }`}
            id={`content${index + 1}`}
          >
            <div className={styles.card}>
              <div className={styles.imgBX}>
                <Image
                  src={member.imageUrl}
                  alt={`recoded${index + 1}`}
                  className={styles.Image}
                />
              </div>
              <div className={styles.textBx}>
                <h2>{member.name}</h2>
                <ul className={styles.sci}>
                  <li>
                    <Link href={member.github}>
                      <FaGithub />
                    </Link>
                  </li>
                  <li>
                    <Link href={member.linkedin}>
                      <AiFillLinkedin />
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        ))}
      </div>
    </div>
  );
}

export default MemberCard;
