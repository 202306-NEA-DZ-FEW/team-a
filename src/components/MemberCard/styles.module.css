.container {
  position: relative;
  width: 580px;
  height: 580px;
  border: 2px solid #fff;
  border-radius: 50%;
  scale: 80%;
}

.container .icon {
  position: relative;
  left: -50%;
  width: 100%;
  height: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
  cursor: pointer;
}

.container .icon .imgbx {
  position: absolute;
  width: 90px;
  height: 90px;
  border-radius: 50%;
  overflow: hidden;
  transition: 0.5s;
  border: 1px solid #fff;
  box-shadow: 0 0 0 4px #222, 0 0 0 6px #fff;
  transform: rotate(calc(360deg / 5 * var(--i)));
  transform-origin: 340px;
  z-index: 10;
}

.container .icon .imgbx.active {
  box-shadow: 0 0 0 6px #222, 0 0 0 10px #f42626;
}

.container .icon .imgbx .Image {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-radius: 50%;
  object-fit: cover;
  transform: rotate(calc(-360deg / 5 * var(--i)));
  transition: 0.5s;
  filter: grayscale(1);
}

.container .icon .imgbx.active .Image {
  filter: grayscale(0);
}

.content {
  position: absolute;
  inset: 0;
  overflow: hidden;
  display: flex;
  justify-content: center;
  align-items: center;
}

.content::before {
  content: "";
  position: absolute;
  inset: 60px;
  border: 4px solid transparent;
  border-left: 4px solid #5db7e7;
  border-right: 4px solid #fff;
  border-radius: 50%;
  animation: animate_01 5s linear infinite;
  z-index: 1;
  pointer-events: none;
}

@keyframes animate_01 {
  0% {
    rotate: 0deg;
  }
  100% {
    rotate: 360deg;
  }
}

.content::after {
  content: "";
  position: absolute;
  inset: 120px;
  border: 4px solid transparent;
  border-left: 4px solid #d9060d;
  border-right: 4px solid #fff;
  border-radius: 50%;
  animation: animate_02 2.5s linear infinite;
  z-index: 1;
  pointer-events: none;
}

@keyframes animate_02 {
  0% {
    rotate: 360deg;
  }
  100% {
    rotate: 0deg;
  }
}

.contentBx {
  position: absolute;
  transform: scale(0);
  transition: 0.5s;
  opacity: 0;
  display: flex;
  justify-content: center;
  align-items: center;
}
.contentBx.active {
  transform: scale(1);
  opacity: 1;
  transition-delay: 0.5s;
}

.contentBx .card {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
  gap: 15px;
}

.contentBx .card .imgBX {
  position: relative;
  width: 150px;
  height: 150px;
  border-radius: 10px;
  overflow: hidden;
}

.contentBx .card .imgBX .Image {
  position: absolute;
  top: 2px;
  left: 2px;
  width: 150px;
  height: 140px;
  object-fit: cover;
}
.contentBx .card .textBx {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
.contentBx .card .textBx h2 {
  position: relative;
  font-size: 1.25em;
  font-weight: 600;
  color: #fff;
  line-height: 1em;
  text-transform: uppercase;
  text-align: center;
}

.contentBx .card .textBx .sci {
  position: relative;
  display: flex;
  gap: 7px;
  margin-top: 10px;
}

/* .contentBx .card .textBx .sci li {
  list-style: none;
} */

.contentBx .card .textBx .sci li a {
  position: relative;
  width: 30px;
  height: 30px;
  display: flex;
  justify-content: center;
  align-items: center;
  background: #444;
  color: #fff;
  text-decoration: none;
  border-radius: 5px;
  transition: 0.5s;
}

.contentBx .card .textBx .sci li a:hover {
  background: #5db7e7;
}

@media (max-width: 405px) {
  .container {
    width: 600px;
    height: 300px;
  }

  .container .icon .imgbx {
    transform-origin: 200px;
    width: 60px;
    height: 60px;
  }

  .content::before {
    inset: 20px;
    border: 2px solid transparent;
    border-left: 3px solid #5db7e7;
    border-right: 3px solid #fff;
  }

  .content::after {
    inset: 40px;
    border: 2px solid transparent;
    border-left: 3px solid #d9060d;
    border-right: 3px solid #fff;
  }
  .contentBx .card {
    gap: 2px;
  }

  .contentBx .card .imgBX .Image {
    margin-top: 24px;
    left: 22px;
    width: 100px;
    height: 100px;
  }
  .contentBx .card .textBx h2 {
    font-size: 1.1em;
    font-weight: 400;
  }
}
