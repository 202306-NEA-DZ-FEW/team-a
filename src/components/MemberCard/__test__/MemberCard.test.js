import image from "public/images/team_1.svg";
import renderer from "react-test-renderer";

import MemberCard from "..";

it("renders correctly", () => {
  const tree = renderer
    .create(
      <MemberCard
        name='name'
        github='https://github.com/'
        linkedin='https://www.linkedin.com/'
        imageUrl={image}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
