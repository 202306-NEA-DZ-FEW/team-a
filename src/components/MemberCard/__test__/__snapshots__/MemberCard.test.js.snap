// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="icon"
  />
  <div
    className="content"
  />
</div>
`;
